.github/workflows/documentation.yaml

```diff
@@ -0,0 +1,30 @@
+name: Build Documentation
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  coverage:
+    runs-on: ubuntu-24.04
+
+    steps:
+    - name: Checkout code including full history and submodules
+      uses: actions/checkout@v4
+      with:
+        submodules: true
```

Comment:
I just checked out without submodules and build the documentation. It does not seem like submodules are actually needed.